datadog/resource_datadog_logs_archive_test.go

```diff
@@ -0,0 +1,343 @@
+						"datadog_logs_archive.my_s3_archive", "name", "my first s3 archive"),
+				),
+			},
+			{
+				Config: archiveS3ConfigForUpdate,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"datadog_logs_archive.my_s3_archive", "name", "my first s3 archive after update"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckArchiveExists(accProvider *schema.Provider) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		providerConf := accProvider.Meta().(*ProviderConfiguration)
+		datadogClientV2 := providerConf.DatadogClientV2
+		authV2 := providerConf.AuthV2
+
+		if err := archiveExistsChecker(authV2, s, datadogClientV2); err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func archiveExistsChecker(authV2 context.Context, s *terraform.State, datadogClientV2 *datadogV2.APIClient) error {
+	for _, r := range s.RootModule().Resources {
+		if r.Type == "datadog_logs_archive" {
+			id := r.Primary.ID
+			if _, _, err := datadogClientV2.LogsArchivesApi.GetLogsArchive(authV2, id).Execute(); err != nil {
+				return fmt.Errorf("received an error when retrieving archive, (%s)", err)
+			}
+		}
+	}
+	return nil
+}
+
+func testAccCheckArchiveAndIntegrationGCSDestroy(accProvider *schema.Provider) func(*terraform.State) error {
+	return func(s *terraform.State) error {
+		err := testAccCheckArchiveDestroy(accProvider)(s)
+		if err != nil {
+			return err
+		}
+		err = checkIntegrationGCPDestroy(accProvider)(s)
+		return err
+	}
+}
+
+func testAccCheckArchiveAndIntegrationAWSDestroy(accProvider *schema.Provider) func(*terraform.State) error {
+	return func(s *terraform.State) error {
+		err := testAccCheckArchiveDestroy(accProvider)(s)
+		if err != nil {
+			return err
+		}
+		err = checkIntegrationAWSDestroy(accProvider)(s)
+		return err
+	}
+}
+
+func testAccCheckArchiveDestroy(accProvider *schema.Provider) func(*terraform.State) error {
+	return func(s *terraform.State) error {
+		providerConf := accProvider.Meta().(*ProviderConfiguration)
+		datadogClientV2 := providerConf.DatadogClientV2
+		authV2 := providerConf.AuthV2
+
+		if err := archiveDestroyHelper(authV2, s, datadogClientV2); err != nil {
+			return err
+		}
+		return nil
+	}
+}
+
+func archiveDestroyHelper(authV2 context.Context, s *terraform.State, datadogClientV2 *datadogV2.APIClient) error {
+	for _, r := range s.RootModule().Resources {
+		if r.Type == "datadog_logs_archive" {
+			id := r.Primary.ID
+			archive, httpresp, err := datadogClientV2.LogsArchivesApi.GetLogsArchive(authV2, id).Execute()
+			if err != nil {
+				if httpresp != nil && httpresp.StatusCode == 404 {
+					continue
+				}
+				return fmt.Errorf("received an error when retrieving pipeline, (%s)", err)
+			}
+			if &archive != nil {
+				return fmt.Errorf("archive still exists")
+			}
+		}
+
+	}
+	return nil
+}
+
+// readFixture opens the file at path and returns the contents as a string
+func readFixture(t *testing.T, path string) string {
```

Comment:
Do we still need fixture now?

Reply:
You're right, this is no longer required. I will clean it up.